python/tk_multi_workfiles/browser_form.py

```diff
@@ -148,25 +168,42 @@ def show_user_filtering_widget(self, is_visible):
 
     def set_models(self, my_tasks_model, entity_models, file_model):
         """
-        Sets the models used by browser.
+        Sets the models used by browser and create the widgets to display them.
 
-        :param my_tasks_model: Instance of the MyTaskModel class
-        :param entity_models: List of ShotgunEntityModel instances.
+        :param my_tasks_model: Instance of the :class:`MyTaskModel`.
+        :param entity_models: List of :class:`ShotgunEntityModel` instances.
         :param file_model: Instance of the file model.
         """
         app = sgtk.platform.current_bundle()
         allow_task_creation = app.get_setting("allow_task_creation")
 
         if my_tasks_model:
             # create my tasks form:
-            self._my_tasks_form = MyTasksForm(my_tasks_model, allow_task_creation, parent=self)
+            self._my_tasks_form = MyTasksForm(
+                my_tasks_model,
+                allow_task_creation,
+                parent=self
+            )
             self._my_tasks_form.entity_selected.connect(self._on_entity_selected)
             self._ui.task_browser_tabs.addTab(self._my_tasks_form, "My Tasks")
             self._my_tasks_form.create_new_task.connect(self.create_new_task)
 
-        for caption, model in entity_models:
-            # create new entity form:
-            entity_form = EntityTreeForm(model, caption, allow_task_creation, [], parent=self)
+        for caption, step_filter_on, model in entity_models:
+            step_entity_filter = None
+            if model.represents_tasks:
```

Comment:
This would not work: in the case of deferred queries the primary `model.get_entity_type()` would return "Asset" (or "Shot"), only the deferred query model would return "Task". Hence, this  extra level of abstraction. Arguably a better name than `represents_tasks` could be found.